### PR TITLE
Optimize `Writer.prototype.escapedValue` for numbers

### PR DESCRIPTION
**Pull request by urain39**

### CODE REVIEW

File: mustache.mjs [hunk 1]
```diff
@@ -650,7 +650,7 @@ Writer.prototype.unescapedValue = function unescapedValue (token, context) {
 Writer.prototype.escapedValue = function escapedValue (token, context) {
   var value = context.lookup(token[1]);
   if (value != null)
-    return mustache.escape(value);
+    return typeof value === 'number' ? String(value) : mustache.escape(value);
```
Comment (urain39) [inline mustache.mjs:653]: This change can reduce about 5~10% time to render.

Comment (phillipj) [reply mustache.mjs:653]: You got any benchmarks on that by any chance?

**Comment by phillipj** [comment]
Thanks for the contribution! Don't mind the failing `deno-usage` test for now, that's not related to these changes and will have to be looked into separately.

File: mustache.mjs (hunk 1 above)
Comment (urain39) [reply mustache.mjs:653]: Not fully tested yet.

Currently i have only tested for two simple cases:

**Case one(values contains number)**:
```js
var mustache = require('./mustache.min');
var mustache2 = require('./mustache.min.new');

mustache.tags = ['{', '}'];
mustache2.tags = ['{', '}'];

var renderData = (function() {
	var datas = [];
	for (var i = 0; i < 100000; i++) {
		datas.push({
			show: i % 2,
			number: i,
			name: String(i),
			url: 'https://' + String(i)
		});
	}

	return datas;
})();

var timer = {
	startTime: 0,
	start: function() {
		this.startTime = new Date().getTime();
	},
	stop: function() {
		return new Date().getTime() - this.startTime;
	},
};

let template = `{#.}
	{#show}
		{number} {name} {url}
	{/show}
	{^show}
		{number} {name} {url}
	{/show}
{/.}
`, template3 = `
{#.}Hello {name}!{/.}
`;


mustache.parse(template);
mustache2.parse(template);


timer.start();
mustache.render(template, renderData);
console.log(`mustache.render took ${timer.stop()}ms`);

timer.start();
mustache2.render(template, renderData);
console.log(`mustache2.render took ${timer.stop()}ms`);
```

**Case two(values not contains number)**:
```ts
...
var renderData = (function() {
	var datas = [];
	for (var i = 0; i < 100000; i++) {
		datas.push({
			show: i % 2,
			number: String(i),
			name: String(i),
			url: 'https://' + String(i)
		});
	}

	return datas;
})();
...
```

Case one log:
```
mustache.render took 2383ms
mustache2.render took 2207ms
mustache.render took 2364ms
mustache2.render took 2200ms
```

Case two log:
```
mustache.render took 2226ms
mustache2.render took 2183ms
mustache.render took 2183ms
mustache2.render took 2176ms
```

By the log, we know that it is works fine on nodejs.

Comment (urain39) [reply mustache.mjs:653]:
```js
var Benchmark = require('benchmark');
var Mustache = require('./mustache.min');
var Mustache2 = require('./mustache2.min');

Mustache.tags = ['{', '}'];
Mustache2.tags = ['{', '}'];

var renderData = (function() {
	var datas = [];
	for (var i = 111111; i < 999999; i++)
		datas.push({ number: i });

	return datas;
})();

var renderData2 = (function() {
	var datas = [];
	for (var i = 111111; i < 999999; i++)
		datas.push({ number: String(i) });

	return datas;
})();

var template = `{number}`;

// Render tests(renderData)
new Benchmark.Suite()
.add('Mustache.render(renderData)', function() {
  Mustache.render(template, renderData);
})
.add('Mustache2.render(renderData)', function() {
  Mustache2.render(template, renderData);
})
// add listeners
.on('cycle', function(event) {
  console.log(String(event.target));
})
.on('complete', function() {
  console.log('(number)Fastest is ' + this.filter('fastest').map('name'));
})
// run async
.run({ 'async': false });


// Render tests(renderData2)
new Benchmark.Suite()
.add('Mustache.render(renderData2)', function() {
  Mustache.render(template, renderData2);
})
.add('Mustache2.render(renderData2)', function() {
  Mustache2.render(template, renderData2);
})
// add listeners
.on('cycle', function(event) {
  console.log(String(event.target));
})
.on('complete', function() {
  console.log('(string)Fastest is ' + this.filter('fastest').map('name'));
})
// run async
.run({ 'async': false });
```

outputs:
```log
Mustache.render(renderData) x 392,991 ops/sec ±1.49% (78 runs sampled)
Mustache2.render(renderData) x 378,421 ops/sec ±0.70% (77 runs sampled)
(number)Fastest is Mustache.render(renderData)
Mustache.render(renderData2) x 381,387 ops/sec ±2.37% (78 runs sampled)
Mustache2.render(renderData2) x 366,697 ops/sec ±5.54% (75 runs sampled)
(string)Fastest is Mustache.render(renderData2),Mustache2.render(renderData2)
```

Comment (urain39) [reply mustache.mjs:653]: It seems that the improvement is not great, i need more testing.

Comment (urain39) [reply mustache.mjs:653]:
```js
var Benchmark = require('benchmark');
var Mustache = require('./mustache.min');
var Mustache2 = require('./mustache2.min');

Mustache.tags = ['{', '}'];
Mustache2.tags = ['{', '}'];

var renderData = (function() {
	var datas = [];
	for (var i = 111111; i < 111999; i++)
		datas.push({ number: i });

	return datas;
})();

var renderData2 = (function() {
	var datas = [];
	for (var i = 111111; i < 111999; i++)
		datas.push({ number: String(i) });

	return datas;
})();

var template = `{#.}{number}{/.}`;

// Render tests(renderData)
new Benchmark.Suite()
.add('Mustache.render(renderData)', function() {
  Mustache.render(template, renderData);
})
.add('Mustache2.render(renderData)', function() {
  Mustache2.render(template, renderData);
})
// add listeners
.on('cycle', function(event) {
  console.log(String(event.target));
})
.on('complete', function() {
  console.log('(number)Fastest is ' + this.filter('fastest').map('name'));
})
// run async
.run({ 'async': false });


// Render tests(renderData2)
new Benchmark.Suite()
.add('Mustache.render(renderData2)', function() {
  Mustache.render(template, renderData2);
})
.add('Mustache2.render(renderData2)', function() {
  Mustache2.render(template, renderData2);
})
// add listeners
.on('cycle', function(event) {
  console.log(String(event.target));
})
.on('complete', function() {
  console.log('(string)Fastest is ' + this.filter('fastest').map('name'));
})
// run async
.run({ 'async': false });
```

**Outputs:**
```log
Mustache.render(renderData) x 527 ops/sec ±9.91% (71 runs sampled)
Mustache2.render(renderData) x 1,755 ops/sec ±1.51% (73 runs sampled)
(number)Fastest is Mustache2.render(renderData)
Mustache.render(renderData2) x 577 ops/sec ±0.98% (76 runs sampled)
Mustache2.render(renderData2) x 567 ops/sec ±1.36% (74 runs sampled)
(string)Fastest is Mustache.render(renderData2),Mustache2.render(renderData2)

Mustache.render(renderData) x 548 ops/sec ±10.31% (72 runs sampled)
Mustache2.render(renderData) x 1,829 ops/sec ±1.24% (77 runs sampled)
(number)Fastest is Mustache2.render(renderData)
Mustache.render(renderData2) x 586 ops/sec ±1.05% (72 runs sampled)
Mustache2.render(renderData2) x 559 ops/sec ±1.54% (73 runs sampled)
(string)Fastest is Mustache.render(renderData2)
```

Comment (urain39) [reply mustache.mjs:653]: Sorry for i made a mistake before : (

**Comment by phillipj** [comment]
> Don't mind the failing deno-usage test for now, that's not related to these changes and will have to be looked into separately.

For future reference; this got fixed in https://github.com/janl/mustache.js/commit/65af14d1e01c74fc94337b9241909d7c153f5cfc

**Pull request merged**